### PR TITLE
Sweeps: Fixes code snippet(s), condenses page

### PR DESCRIPTION
**Pull request by ngrayluna**
Description
-----------
Started off with a simple code check, evolved to rewriting most of this page.

Related issues
-----------
- Fixes https://wandb.atlassian.net/browse/DOCS-1695

📄 **[View preview links for changed pages](https://github.com/wandb/docs/pull/1518#issuecomment-3211858307)**

### CODE REVIEW

File: content/en/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 1]
```diff
@@ -8,19 +8,27 @@ title: Add W&B (wandb) to your code
...
+Suppose you have a Python script that trains a model (see below). Your goal is to find the hyperparameters that maxmimizes the validation accuracy(`val_acc`).
 
-We defined a dictionary called `config` where we store hyperparameters values. At the end of the cell, we call the `main` function to execute the mock training code.
+In your Python script, you define two functions: `train_one_epoch` and `evaluate_one_epoch`. The `train_one_epoch` function simulates training for one epoch and returns the training accuracy and loss. The `evaluate_one_epoch` function simulates evaluating the model on the validation data set and returns the validation accuracy and loss.
+
+You define a configuration dictionary (`config`) that contains hyperparameter values such as the learning rate (`lr`), batch size (`batch_size`), and number of epochs (`epochs`). The values in the configuration dictionary are used to control the training process.
```
Comment (mdlinville) [inline content/en/guides/models/sweeps/add-w-and-b-to-your-code.md]:
```suggestion
You define a configuration dictionary (`config`) that contains hyperparameter values such as the learning rate (`lr`), batch size (`batch_size`), and number of epochs (`epochs`). The values in the configuration dictionary control the training process. 
```

File: content/en/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 2]
```diff
@@ -8,19 +8,27 @@ title: Add W&B (wandb) to your code
...
+In your Python script, you define two functions: `train_one_epoch` and `evaluate_one_epoch`. The `train_one_epoch` function simulates training for one epoch and returns the training accuracy and loss. The `evaluate_one_epoch` function simulates evaluating the model on the validation data set and returns the validation accuracy and loss.
+
+You define a configuration dictionary (`config`) that contains hyperparameter values such as the learning rate (`lr`), batch size (`batch_size`), and number of epochs (`epochs`). The values in the configuration dictionary are used to control the training process. 
+
+Next you define a function called `main` that mimics a typical training loop. For each epoch, the accuracy and loss is computed on the training and validation data sets.
+
+{{< alert >}}
+This code is a mock training script. It does not train a model, but simulates the training process by generating random accuracy and loss values. The purpose of this code is to demonstrate how to integrate W&B into your training script.
```
Comment (mdlinville) [inline content/en/guides/models/sweeps/add-w-and-b-to-your-code.md:24]: I like "integrate W&B into your training script" more than the way you word it in the overview, FWIW.

File: content/en/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 3]
```diff
@@ -31,129 +39,127 @@ def evaluate_one_epoch(epoch):
...
         print("training accuracy:", train_acc, "training loss:", train_loss)
-        print("validation accuracy:", val_acc, "training loss:", val_loss)        
+        print("validation accuracy:", val_acc, "validation loss:", val_loss)
+
+if __name__ == "__main__":
+    main()
 ```
 
-### Training script with W&B Python SDK
+In the next section, you will add W&B to your Python script to track hyperparameters and metrics during training. You want to use W&B to find the best hyperparameters that maximize the validation accuracy (`val_acc`).
```
Comment (mdlinville) [inline content/en/guides/models/sweeps/add-w-and-b-to-your-code.md:61]:
```suggestion
The next section shows how to add W&B to your Python script so it can track hyperparameters and metrics during training, and how W&B can help you find the hyperparameters that produce the best validation accuracy (`val_acc`).
```
Ignore if this is introducing inaccuracies.

File: content/en/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 4]
```diff
@@ -31,129 +39,127 @@ def evaluate_one_epoch(epoch):
...
 
-The following code examples demonstrate how to add the W&B Python SDK into your
-code. If you start W&B Sweep jobs in the CLI, you will want to explore the CLI
-tab. If you start W&B Sweep jobs within a Jupyter notebook or Python script,
-explore the Python SDK tab.
+## Training script with W&B Python SDK
 
-{{< tabpane text=true >}} {{% tab header="Python script or notebook" %}} To
-create a W&B Sweep, we added the following to the code example:
+How you integrate W&B to your Python script or notebook depends on how you manage sweeps. You can start a sweep job within a Python notebook or script or from the command line.
```
Comment (mdlinville) [inline content/en/guides/models/sweeps/add-w-and-b-to-your-code.md:66]: I'd pick either "notebook or script" or "script or notebook" here and in the tabs below.

File: content/en/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 5]
```diff
@@ -31,129 +39,127 @@ def evaluate_one_epoch(epoch):
...
-4. Use the [`wandb.init()`]({{< relref "/ref/python/sdk/functions/init.md" >}}) API to generate a background process to sync and log data as a [W&B Run]({{< relref "/ref/python/sdk/classes/run.md" >}}).
-5. (Optional) define values from `wandb.config` instead of defining hard coded values.
-6. Log the metric you want to optimize with [`wandb.Run.log()`]({{< relref "/ref/python/sdk/classes/run.md/#method-runlog" >}}). You must log the metric defined in your configuration. Within the configuration dictionary (`sweep_configuration` in this example), you define the sweep to maximize the `val_acc` value.
-7. Start the sweep with the [`wandb.agent`]({{< relref "/ref/python/sdk/functions/agent.md" >}}) API call. Provide the sweep ID and the name of the function the sweep will execute (`function=main`), and specify the maximum number of runs to try to four (`count=4`). For more informationp, see [Start sweep agents]({{< relref "./start-sweep-agents.md" >}}).
+{{< tabpane text=true >}}
+{{% tab header="Python script or notebook" %}} 
 
+Add the following to your Python script:
```
Comment (mdlinville) [inline content/en/guides/models/sweeps/add-w-and-b-to-your-code.md:71]:
```suggestion
Make some adjustments to your Python script:
```

File: content/en/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 6]
```diff
@@ -31,129 +39,127 @@ def evaluate_one_epoch(epoch):
...
+
+1. Create a dictionary object where the key-value pairs define a [sweep configuration]({{< relref "/guides/models/sweeps/define-sweep-configuration/" >}}). The sweep configuration defines the hyperparameters you want W&B to explore on your behalf along with the metric you want to optimize. Continuing from the previous example, the batch size (`batch_size`), epochs (`epochs`), and the learning rate (`lr`) are the hyperparameters to vary during each sweep. You want to maximize the accuracy of the validation score so you set `"goal": "maximize"` and the name of the variable you want to optimize for, in this case `val_acc` (`"name": "val_acc"`).
```
Comment (mdlinville) [inline content/en/guides/models/sweeps/add-w-and-b-to-your-code.md:73]: Consider moving the "Continuing from...." up before the steps start. It makes this step look disproportionately long.

File: content/en/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 7]
```diff
@@ -31,129 +39,127 @@ def evaluate_one_epoch(epoch):
...
+2. Pass the sweep configuration dictionary to [`wandb.sweep()`]({{< relref "/ref/python/sdk/functions/sweep.md" >}}). This initializes the sweep and returns a sweep ID (`sweep_id`). For more information, see [Initialize sweeps]({{< relref "./initialize-sweeps.md" >}}).
+3. At the top of your script, import the W&B Python SDK (`wandb`).
```
Comment (mdlinville) [inline content/en/guides/models/sweeps/add-w-and-b-to-your-code.md:75]: This seems out of order. Wouldn't you already need it to invoke wandb.sweep()?

File: content/en/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 8]
```diff
@@ -31,129 +39,127 @@ def evaluate_one_epoch(epoch):
...
+4. Within your `main` function, use the [`wandb.init()`]({{< relref "/ref/python/sdk/functions/init.md" >}}) API to generate a background process to sync and log data as a [W&B Run]({{< relref "/ref/python/sdk/classes/run.md" >}}). Pass the project name as a parameter to the `wandb.init()` method. If you do not pass a project name, W&B uses the default project name.
```
Comment (mdlinville) [inline content/en/guides/models/sweeps/add-w-and-b-to-your-code.md:76]: Is it worth showing how to find the default project?

File: content/en/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 9]
```diff
@@ -31,129 +39,127 @@ def evaluate_one_epoch(epoch):
...
+5. Fetch the hyperparameter values from the `wandb.Run.config` object. This allows you to use the hyperparameter values defined in the sweep configuration dictionary instead of hard coded values.
+6. Log the metric you are optimizing for to W&B using [`wandb.Run.log()`]({{< relref "/ref/python/sdk/classes/run.md/#method-runlog" >}}). You must log the metric defined in your configuration. For example, if you define the metric to optimize as `val_acc`, you must log `val_acc`. If you do not log the metric, W&B does not know what to optimize for. Within the configuration dictionary (`sweep_configuration` in this example), you define the sweep to maximize the `val_acc` value.
+7. Start the sweep with [`wandb.agent()`]({{< relref "/ref/python/sdk/functions/agent.md" >}}). Provide the sweep ID and the name of the function the sweep will execute (`function=main`), and specify the maximum number of runs to try to four (`count=4`).
+
+
+Putting this all together, your script might look simlar to the following:
```
Comment (mdlinville) [inline content/en/guides/models/sweeps/add-w-and-b-to-your-code.md]:
```suggestion
Putting this all together, your script might look similar to the following:
```

File: content/en/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 10]
```diff
@@ -258,84 +264,34 @@ For more information, see [Start sweep jobs]({{< relref "./start-sweep-agents.md
 
 {{% /tab %}} {{< /tabpane >}}
 
-## Consideration when logging metrics
 
-Be sure to log the sweep's metric to W&B explicitly. Do not log metrics for your sweep inside a subdirectory.
-
-For example, consider the following pseudocode. A user wants to log the validation loss (`"val_loss": loss`). First they pass the values into a dictionary. However, the dictionary passed to `wandb.Run.log()` does not explicitly access the key-value pair in the dictionary:
+{{% alert title="Logging metrics to W&B in a sweep"%}}
```
Comment (mdlinville) [inline content/en/guides/models/sweeps/add-w-and-b-to-your-code.md]:
```suggestion
{{% alert title="Logging metrics to W&B in a sweep" %}}
```
Hugo is really picky about whitespace here

File: content/en/guides/models/sweeps/add-w-and-b-to-your-code.md [hunk 11]
```diff
@@ -258,84 +264,34 @@ For more information, see [Start sweep jobs]({{< relref "./start-sweep-agents.md
...
-    "parameters": {
-        "x": {"max": 0.1, "min": 0.01},
-        "y": {"values": [1, 3, 7]},
-    },
-}
-
-sweep_id = wandb.sweep(sweep=sweep_configuration, project="my-first-sweep")
-
-wandb.agent(sweep_id, function=main, count=10)
+```python
```
Comment (mdlinville) [inline content/en/guides/models/sweeps/add-w-and-b-to-your-code.md:284]: You say above that it won't work, but it is worth also saying it in a comment here, and consider disabling the copy button. (I don't remember how)

**Review by mdlinville — APPROVED** [review]
Some minor things that I leave up to you whether to address or follow up or opt against.
